pallets/lbp/src/lib.rs

```diff
@@ -513,7 +530,17 @@ pub mod pallet {
 				pool.final_weight = final_weight.unwrap_or(pool.final_weight);
 
 				pool.fee = fee.unwrap_or(pool.fee);
-				pool.fee_collector = fee_collector.unwrap_or_else(|| pool.fee_collector.clone());
+
+				// Handle update of fee collector - validate and replace old fee collector
+				if let Some(updated_fee_collector) = fee_collector {
+					Self::validate_fee_collector(updated_fee_collector.clone(), pool.assets.0)?;
+
+					<FeeCollectorWithAsset<T>>::remove(pool.fee_collector.clone(), pool.assets.0);
+					<FeeCollectorWithAsset<T>>::insert(updated_fee_collector.clone(), pool.assets.0, true);
+
```

Comment:
With the following refactor, you get one less storage read ( the check in validate).

Also note that the clone calls have disappeared. there is no need to do that. 

And this way, you get the benefit of ValueQuery too.

```suggestion
					FeeCollectorWithAsset::<T>::try_mutate(
						&updated_fee_collector,
						pool.assets.0,
						|collector| -> DispatchResult {
							ensure!(!*collector, Error::<T>::FeeCollectorWithAssetAlreadyUsed);

							<FeeCollectorWithAsset<T>>::remove(&pool.fee_collector, pool.assets.0);
							*collector = true;

							Ok(())
						},
					)?;

```